packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts

```diff
@@ -390,25 +308,7 @@ export function createInsightsMiddleware<
         // This updates userToken which is set explicitly by `aa('setUserToken', userToken)`
         insightsClient(
           'onUserTokenChange',
-          (token) => setUserTokenToSearch(token, 'default', true),
-          {
-            immediate: true,
-          }
```

Comment:
Not sure about this removal, if my previous comment holds true.

Reply:
this part is still present actually, the diff is a little confusing, but `{ immediate: true }` is a parameter to the insights client, while `setUserTokenToSearch(token, token, false)` (or `setUserToken(token)`) is the timeout branch